chain/consensus/mir/mempool/fifo/fifo_pool.go

```diff
@@ -0,0 +1,49 @@
+package fifo
+
+import (
+	mirrequest "github.com/filecoin-project/mir/pkg/pb/requestpb"
+)
+
+// Pool is a structure to implement the simplest pool that enforces FIFO policy on client messages.
+// When a client sends a message we add clientID to orderingClients map and clientByCID.
+// When we receive a message we find the clientID and remove it from orderingClients.
+// We don't need using sync primitives since the pool's methods are called only by one goroutine.
+type Pool struct {
+	clientByCID     map[string]string // messageCID -> clientID
+	orderingClients map[string]bool   // clientID -> bool
+}
+
+func New() *Pool {
+	return &Pool{
+		clientByCID:     make(map[string]string),
+		orderingClients: make(map[string]bool),
+	}
+}
+
+// AddRequest adds the request if it satisfies to the FIFO policy.
+func (p *Pool) AddRequest(cid string, r *mirrequest.Request) (exist bool) {
```

Comment:
> Also, the client IDs could probably be mirtype.ClientID...

Because ClientID in Mir's Request is a string. So there is not much sense to use NodeID here. This only adds the Mir as a dependency into the pool package

Reply:
> unless we assume that requests are being added in contiguous nonce order.

@adlrocha I thought we do. Is that correct?

Reply:
We do, this is handled by Lotus. The mempool will serve user requests in increasing order, so that works.

Feel free to resolve if there are no outstanding concerns. Thanks!